lib/Dialect/AIEVec/Transforms/ConvertVectorToAIEVec.cpp

```diff
@@ -284,51 +310,46 @@ struct ConvertMulToAIEVecMulElemOpPattern
     // %1 = arith.extsi %a : vector<32xi8> to vector<32xi32>
     // %2 = arith.extsi %b : vector<32xi8> to vector<32xi32>
     // %3 = arith.muli %1, %2 : vector<32xi32>
+    // or
+    // %1 = arith muli %a, %b : vector<32xi8>
     // to -
     // aievec.mul_elem(%a, %b) : vector<64xi8>, vector<64xi8>, vector<32xi32>
-    if (laneSize == 32 && resultElWidth == 32) {
-      auto lhs = dyn_cast<arith::ExtSIOp>(mulOp->getOperand(0).getDefiningOp());
-      auto rhs = dyn_cast<arith::ExtSIOp>(mulOp->getOperand(1).getDefiningOp());
-
-      if (!lhs || !rhs)
-        return failure();
-
-      auto lval = lhs->getOperand(0);
-      auto rval = rhs->getOperand(0);
-
-      VectorType lSrcType = cast<VectorType>(lval.getType());
-      VectorType rSrcType = cast<VectorType>(rval.getType());
-
-      unsigned lBitWidth = lSrcType.getElementType().getIntOrFloatBitWidth();
-      unsigned rBitWidth = rSrcType.getElementType().getIntOrFloatBitWidth();
-
-      if (lBitWidth != 8 || rBitWidth != 8)
-        return failure();
-
-      Type accType = getVectorOpDestType(lSrcType, /*AIEML =*/true);
-      VectorType vecType =
-          createVectorType(512 / lBitWidth, lSrcType.getElementType());
-
-      auto zeroConstOp = rewriter.create<arith::ConstantOp>(
-          lhs.getLoc(), rewriter.getIntegerAttr(lSrcType.getElementType(), 0));
-
-      auto broadcastZeroOp = rewriter.create<aievec::BroadcastScalarOp>(
-          lhs.getLoc(), vecType, zeroConstOp->getResult(0));
-      auto extOp = rewriter.create<aievec::ExtOp>(
-          lhs.getLoc(), lSrcType, broadcastZeroOp.getResult(), 0);
-
-      SmallVector<Value> lSources = {lval, extOp->getResult(0)};
-      SmallVector<Value> rSources = {rval, extOp->getResult(0)};
-      auto lConcatOp =
-          rewriter.create<aievec::ConcatOp>(lhs.getLoc(), vecType, lSources);
-      auto rConcatOp =
-          rewriter.create<aievec::ConcatOp>(rhs.getLoc(), vecType, rSources);
-
-      auto mulElemOp = rewriter.create<aievec::MulElemOp>(
-          mulOp.getLoc(), accType, lConcatOp->getResult(0),
-          rConcatOp->getResult(0));
-      rewriter.replaceOpWithNewOp<aievec::CastOp>(
-          mulOp, resultType, mulElemOp.getResult(), /*isResAcc*/ false);
+    if (laneSize == 32 && (resultElWidth == 32 || resultElWidth == 8)) {
+      if (resultElWidth == 32) {
+        auto lhs =
+            dyn_cast<arith::ExtSIOp>(mulOp->getOperand(0).getDefiningOp());
+        auto rhs =
+            dyn_cast<arith::ExtSIOp>(mulOp->getOperand(1).getDefiningOp());
```

Comment:
Beware, you're using the op, not the adaptor here. Also, it's preferable to use the standard op getters rather than the ODS methods. So, `adaptor.getLhs()` and `adaptor.getRhs()`